js/mirador_viewer.js

```diff
@@ -12,10 +12,13 @@
      */
     Drupal.behaviors.Mirador = {
         attach: function (context, settings) {
+            settings.mirador.instances = {}
             Object.entries(settings.mirador.viewers).forEach(entry => {
-              const [base, values] = entry;
-              once('mirador-viewer', base, context).forEach(() =>
-                Mirador.viewer(values, window.miradorPlugins || {})
+              const [viewerId, settings] = entry;
+              once('mirador-viewer', viewerId, context).forEach(() =>
+                // save the mirador instance so other modules can interact
+                // with the store/actions at e.g. drupalSettings.mirador.instances["#mirador-xyz"].store
+                settings.mirador.instances[viewerId] = Mirador.viewer(settings, window.miradorPlugins || {})
```

Comment:
Not sure where to comment on this exactly, but not sure that putting this into `drupalSettings` is the best approach, but instead might make sense to define in a different scope? A `Drupal.IslandoraMirador.instances = {}` or something of the like to which we add? Just thinking: `drupalSettings` is more expected to be managed by Drupal, with stuff merged in automatically with Drupal's attachement mechanism. 

Also, if this attachment happened to be called multiple times in a rendering of the page, it might lose the reference to the instances, overwriting with an empty object? Might have to do something like:

```js
            settings.mirador.instances = settings.mirador.instances || {}
```

for whatever final variable name.

Reply:
Also, should probably have something on the `detach` side, to avoid potentially leaking instances. Would probably have to be conditional on the return from `once.remove()`?